components/SpotifyInterface/SpotifyInterface.c

```diff
@@ -0,0 +1,277 @@
+#include <stdio.h>
+#include "SpotifyInterface.h"
+
+// ****************************** Local Variables
+static const char *TAG = "SpotifyTask";
+
+// ****************************** Local Functions
+static void Spotify_GetToken(char *code, size_t SizeOfCode);
+static bool Spotify_IsTokenValid(void);
+static void Spotify_RenewAccessToken(Token_t *token);
+static void Spotify_MainTask(void *pvparameters);
+
+// ******************************
+SpotifyPrivateHandler_t PrivateHandler;
+SpotifyInterfaceHandler_t InterfaceHandler;
+EventHandlerDataStruct_t EventHandlerData;
+HttpLocalServerParam_t HttpLocalServerParam;
+static httpd_handle_t SpotifyLocalServer = NULL;
+
+// ******************************
+
+ESP_EVENT_DECLARE_BASE(BASE_SPOTIFY_EVENTS);
+
+/**
+ * @brief This function initiates the Spotify authorization process.
+ * @param SpotifyInterfaceHandler as the handler
+ * @return true if task run to the end
+ */
+bool Spotify_TaskInit(SpotifyInterfaceHandler_t *SpotifyInterfaceHandler, uint16_t SpotifyTaskStackSize)
+{
+    InterfaceHandler = *SpotifyInterfaceHandler;
+    PrivateHandler.status = IDLE;
+    if (InterfaceHandler.CheckAddressInSpiffs != NULL &&
+        InterfaceHandler.ReadTxtFileFromSpiffs != NULL &&
+        InterfaceHandler.WriteTxtFileToSpiffs != NULL &&
+        InterfaceHandler.ConfigAddressInSpiffs != NULL &&
+        InterfaceHandler.HttpsResponseReadySemaphore != NULL &&
+        InterfaceHandler.IsSpotifyAuthorizedSemaphore != NULL &&
+        InterfaceHandler.EventHandlerCallBackFunction != NULL &&
+        InterfaceHandler.HttpsBufQueue != NULL)
+    {
+        xTaskCreate(&Spotify_MainTask, "Spotify_MainTask", SpotifyTaskStackSize, NULL, 9, NULL);
+        ESP_LOGI(TAG, "Spotify app initiated successfully");
+    }
+    else
+    {
+        ESP_LOGE(TAG, "SpotifyIntefaceHandler is is missing some pointers, can not run the app");
+        return false;
+    }
+    return true;
+}
+
+/**
+ * @brief Run Http local service
+ */
+void HttpServerService()
```

Comment:
is it the init function? if yes add Init to end of the name

Reply:
I change it to HttpServerServiceInit()